0.10/s2i/bin/assemble

```diff
@@ -47,8 +47,51 @@ if [ -n "$NPM_MIRROR" ]; then
 	npm config set registry $NPM_MIRROR
 fi
 
-echo "---> Building your Node application from source"
-npm install
+# Set the DEV_MODE to false by default.
+if [ -z "$DEV_MODE" ]; then
+  export DEV_MODE=false
+fi
+
+# If NODE_ENV is not set by the user, then NODE_ENV is determined by whether
+# the container is run in development mode.
+if [ -z "$NODE_ENV" ]; then
+  if [ "$DEV_MODE" == true ]; then
+    export NODE_ENV=development
+  else
+    export NODE_ENV=production
+  fi
+fi
+
+if [ "$NODE_ENV" != "production" ]; then
+
+	echo "---> Building your Node application from source"
+	npm install
+
+else
+
+	echo "---> Installing all dependencies"
+	NODE_ENV=development npm install
```

Comment:
Might be clearer what's happening if this was changed to `npm install --only=dev`

Reply:
The npm 2.15.1 (used in centos/nodejs-4-centos7 container) does not support `--only=prod|dev` option.

Reply:
Ah, nice find. And `NODE_ENV=development` is probably less likely to affect `postinstall` that relies on all deps being present 👍